analyzers/src/SonarAnalyzer.Common/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -39,6 +41,22 @@ public static bool IsGenerated(this SyntaxTree tree, GeneratedCodeRecognizer gen
         // Hotpath: Don't use cache.GetOrAdd that takes a factory method. It allocates a delegate which causes GC preasure.
         return cache.TryGetValue(tree, out var isGenerated)
             ? isGenerated
-            : cache.GetOrAdd(tree, generatedCodeRecognizer.IsConsideredGenerated(tree));
+            : cache.GetOrAdd(tree, generatedCodeRecognizer.IsGenerated(tree));
+    }
+
+    public static bool IsConsideredGenerated(this SyntaxTree tree, GeneratedCodeRecognizer generatedCodeRecognizer, Compilation compilation) =>
```

Comment:
@csaba-sagi-sonarsource @cristian-ambrosini-sonarsource I guess this should be `[PerformanceSensitive]` as the one above?

Is `IsGenerated` used anymore outside of this class? Should it still be public or should it be made private, to disallow using it by mistake instead of `IsConsideredGenerated`?

Reply:
`IsGenerated` is performance-sensitive because it is on a hotPath and as the comment there says it was using a method which was allocating a lot of delegates. Since `IsConsideredGenerated` does not do any magic but only more or less class the other method I did not see it necessary to put the attribute on it. + This is not the final version of this method, it will change for sure once we implement the proper logic that uses the `sonar.cs.EnableRazorAnalysis` parameter instead of the environment variable.

Regarding the accessibility of `IsGenerated`, it is already used in 1 place outside of this class, and there is [this](https://github.com/SonarSource/sonar-dotnet/issues/7858) issue, that suggest to use it in more places to get the advantage of the caching logic.